### ISSUE TITLE
Can you explain this line of code?

### ISSUE THREAD

**Issue by TobyGE** [apoddar573/Tic-Tac-Toe-Gym_Environment#2]
Thank you for your code! It's really helpful for rl beginners.
In your code, I don't understand what does this line stand for
            metadata = {'render.modes': ['human']}
Can you explain to me? Thx.

**Comment by apoddar573** [comment]
Hi, I am not really sure about this line but this seems to be one of the default lines we need we put while making our custom environment. 
One thing I can suggest you to do is to remove this line and try to make a custom environment and see if it works that way and if it does, does it behave any differently. Please share the results with us here if you do try that.

**Comment by TobyGE** [comment]
I removed it, and the code still worked.

**Comment by apoddar573** [comment]
Okay, so I did some more digging and I might have found the reason. A snippet from their documentation [here](https://mushroomrl.readthedocs.io/en/latest/_modules/gym/core.html) shows the following comment. 

`Example:

        class MyEnv(Env):
            metadata = {'render.modes': ['human', 'rgb_array']}

            def render(self, mode='human'):
                if mode == 'rgb_array':
                    return np.array(...) # return RGB frame suitable for video
                elif mode == 'human':
                    ... # pop up a window and render
                else:
                    super(MyEnv, self).render(mode=mode) # just raise an exception
        `
Basically this is a way of rendering different representations of the state in different modes. The metadata line you asked about just defines the different rendering modes available.

Hope my answer helps. Thanks for helping me learn something new too! :)

**Comment by TobyGE** [comment]
Thank you so much~
